[0035] Relax groupshared load/store requirements (#904)
microsoft/hlsl-specs: proposals/0035-linalg-matrix.md

                   MatrixLayoutEnum Layout);

template <typename T, SIZE_TYPE Size>
static typename hlsl::enable_if<
    (hlsl::is_same<typename hlsl::strip_vector_type<T>::type,
                   ElementType>::value ||
     hlsl::is_integral_or_packed_integral<
         typename hlsl::strip_vector_type<T>::type>::value),

>     Matrix>::type

Reviewer: ```suggestion
    void>::type
```

The matrix `Store` methods store the matrix data to a target
`RWByteAddressBuffer` or `groupshared` array. When storing to
`RWByteAddressBuffer` objects the data is stored in the component type of the
matrix object. When storing to `groupshared` memory, the array may be of any
arithmetic or packed data type. The type of the array must match the type of the

> matrix being stored into the array, or the array it must be an integer or packed

Reviewer: I think there is a typo here but I'm not exactly sure where. My brain is getting tripped up on `, or the array it must be an integer or`

Author: This is how you know I'm not using copilot to write these... pretty sure even a bad LLM can English better than this 😝

> matrix being stored into the array, or the array it must be an integer or packed

Author: ```suggestion
matrix being stored into the array, or the array must be of integer or packed
```
